pandas/core/arrays/masked.py

```diff
@@ -156,7 +161,9 @@ def to_numpy(
         if na_value is lib.no_default:
             na_value = libmissing.NA
         if dtype is None:
-            dtype = object
+            dtype = np.dtype(object)
+        else:
+            dtype = pandas_dtype(dtype)
```

Comment:
I don't think we should do this here, as this is converting to a numpy array, the `dtype` can never be a pandas dtype, and is always meant to be a numpy dtype (so let numpy interpret the string / dtype-like objects)

Reply:
ok